### PR TITLE
Get-Content -Raw should not neglects to read the last character

### LINKED ISSUE #4980: Get-Content -Raw neglects to read the last character, if it is a LF
Steps to reproduce
------------------

Save the following to `t.Tests.ps1` and invoke with `./t.Tests.ps1`:

```powershell
Describe "Get-Content -Raw test" {
    BeforeAll {
        Push-Location TestDrive:/
        $lfTextUnterminated = "a`nb"
        $lfText = "a`nb`n"
        $crlfTextUnterminated = "a`r`nb"
        $crlfText = "a`r`nb`r`n"
        Set-Content -Encoding Ascii -NoNewline 'lf.txt' -value $lfText
        Set-Content -Encoding Ascii -NoNewline 'crlf.txt' -value $crlfText
        Set-Content -Encoding Ascii -NoNewline 'lf-nt.txt' -value $lfTextUnterminated
        Set-Content -Encoding Ascii -NoNewline 'crlf-nt.txt' -value $crlfTextUnterminated
    }
    It "Reads LF-terminated files in full." {
        Get-Content -Raw 'lf.txt' | Should BeExactly $lfText
    }
    It "Reads CRLF-terminated files in full." {
        Get-Content -Raw 'crlf.txt' | Should BeExactly $crlfText
    }
    It "Reads LF-separated files without trailing newline in full." {
        Get-Content -Raw 'lf-nt.txt' | Should BeExactly $lfTextUnterminated
    }
    It "Reads CRLF-separated files without trailing newline in full." {
        Get-Content -Raw 'crlf-nt.txt' | Should BeExactly $crlfTextUnterminated
    }
    AfterAll {
        Remove-Item -LiteralPath 'lf.txt', 'lf-nt.txt', 'crlf.txt', 'crlf-nt.txt'
        Pop-Location
    }
}
```

Expected behavior
-----------------

All tests should pass.

Actual behavior
---------------

The files ending in an LF character are read without that last character.
Note that Windows Powershell is _not_ affected.

```none
...
 [-] Reads LF-terminated files in full. 47ms
   Expected string length 4 but was 3. Strings differ at index 3.
   Expected: {a\nb\n}
   But was:  {a\nb}
   ---------------^
   at line: 18 in /Users/mklement/Desktop/pg/t.Tests.ps1
   18:         Get-Content -Raw 'lf.txt' | Should BeExactly $lfText
 [-] Reads CRLF-terminated files in full. 20ms
   Expected string length 6 but was 5. Strings differ at index 5.
   Expected: {a\r\nb\r\n}
   But was:  {a\r\nb\r}
   -------------------^
   at line: 21 in /Users/mklement/Desktop/pg/t.Tests.ps1
   21:         Get-Content -Raw 'crlf.txt' | Should BeExactly $crlfText
...
```

Environment data
----------------

```powershell
PowerShell Core v6.0.0-beta.7 on macOS 10.12.6
PowerShell Core v6.0.0-beta.7 on Ubuntu 16.04.3 LTS
PowerShell Core v6.0.0-beta.7 on Microsoft Windows 10 Pro (64-bit; v10.0.15063)
```

### PR DESCRIPTION
**Pull request by chunqingchen**
resolve #4980

once the -raw parameter is set, we should force the contentString not to substring under any circumstance

Before the fix:
get-content -raw will ignore \n 

After the fix:
get-content -raw will get all the contents include \n

### CODE REVIEW

**Comment by iSazonov** [comment]
@mklement0 Could you please review?

File: src/System.Management.Automation/namespaces/FileSystemContentStream.cs [hunk 1]
```diff
@@ -733,7 +733,7 @@ private bool ReadDelimited(bool waitChanges, ArrayList blocks, bool readBackward
                 //    (Trimming it during backward reading would not only be unnecessary, but could interfere with determining the correct start position.)
                 string contentString = content.ToString();
                 blocks.Add(
-                    !readBackward && contentString.EndsWith(actualDelimiter, StringComparison.Ordinal)
+                    !readBackward && contentString.EndsWith(actualDelimiter, StringComparison.Ordinal) && !_isRawStream
```
Comment (anmenaga) [inline src/System.Management.Automation/namespaces/FileSystemContentStream.cs]: This looks good;
but original issue says that "Note that Windows Powershell is not affected".
Any ideas why this does not repro in Windows PS?

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 [hunk 2]
```diff
@@ -285,3 +285,28 @@ baz
         }
     }
 }
+
+Describe "Get-Content -Raw test" -Tags "CI" {
```
Comment (anmenaga) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: This test code is a good candidate to be rewritten using `-testcases` construct, as per [testing guidelines](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md#parameter-generation).

**Review by anmenaga — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/namespaces/FileSystemContentStream.cs (hunk 1 above)
Comment (chunqingchen) [reply src/System.Management.Automation/namespaces/FileSystemContentStream.cs]: This is because the code is introduced by core ps code change.

**Comment by chunqingchen** [comment]
@anmenaga your comments are resolved. thanks

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 [hunk 3]
```diff
@@ -287,26 +287,17 @@ baz
...
-    }
-    It "Reads CRLF-separated files without trailing newline in full." {
-        Get-Content -Raw "$TestDrive\crlf-nt.txt" | Should BeExactly $crlfTextUnterminated
+
+    $testcases = @(
+                  @{character = "a`nb`n"; testname = "LF-terminated files"; filename = "lf.txt"}
+                  @{character = "a`r`nb`r`n"; testname = "CRLF-terminated files"; filename = "crlf.txt"}
+                  @{character = "a`nb"; testname = "LF-separated files without trailing newline"; filename = "lf-nt.txt"}
+                  @{character = "a`r`nb"; testname = "CRLF-separated files without trailing newline"; filename = "crlf-nt.txt"}
+                )
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: Please move to `It` block:
```powershell
It "Reads - <testname> in full" -TestCases @(
    @{character = "a`nb`n"; testname = "LF-terminated files"; filename = "lf.txt"} 
    ....
) {
    ....
}

**Review by anmenaga — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 (hunk 3 above)
Comment (chunqingchen) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: Thanks for your comment. But I would rather not change it as it is just format preference and the submitted format is following the [testing guidelines](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md#parameter-generation) example.

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: It is new code not formatting. We should place it in a right place. The right place is either `It` block ([sample](https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Modules/Microsoft.PowerShell.Management/ControlService.Tests.ps1#L12)) or `BeforeAll` block. We use the set only one time so `It` block is best place. This does not contradict the testing guidelines.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@chunqingchen Thanks for the fix!

**Pull request merged**